### PR TITLE
Print benchmark results even if it's stopped early via ctrl-c

### LINKED ISSUE #56: If a benchmark is stopped early, it should print results so far

### PR DESCRIPTION
**Pull request by prashantv**
Move the run limiter logic into a separate package with a Stop method.

Fixes #56

### CODE REVIEW

File: benchmark.go [hunk 1]
```diff
@@ -163,3 +134,16 @@ func runBenchmark(out output, allOpts Options, m benchmarkMethod) {
 	out.Printf("Total requests:    %v\n", len(overall.latencies))
 	out.Printf("RPS:               %.2f\n", float64(len(overall.latencies))/total.Seconds())
 }
+
+// stopOnInterrupt sets up a signal that will trigger the run to stop.
+func stopOnInterrupt(out output, r *limiter.Run) {
```
Comment (breerly) [inline benchmark.go]: nice

**Comment by breerly** [comment]
* having trouble following why the rate limiter is in this PR
* don't see any tests issuing `os.Interrupt`

**Comment by abhinav** [comment]
+1 on moving the rate limiter changes into a separate PR

**Comment by prashantv** [comment]
No tests for `os.Interrupt` since sending a `SIGINT` stops the process. It's possible to test by spawning a new process, but I don't think it's worth the effort.

The rate limiter changes were the main thing required to Stop (adding the Stop method), but I wanted to add tests and move it out to clean up the code, moved the refactor into a separate PR though: #59 

Will rebase this change once that lands.

**Comment by prashantv** [comment]
Rebased + updated, PTAL

**Comment by abhinav** [comment]
LGTM

**Pull request merged**